20231214-listUsers-api.md

```diff
@@ -0,0 +1,887 @@
+      resolved_paths: [
+        {
+          "object": "document:example",
+          "relation": "viewer"
+        }
+      ]
+    }
+  ],
+  wildcard_types: ["user"]
+}
+```
+Two things are important to point out about this response:
+
+1. We didn't return `user:will` twice, because once we find the group `group:engineering#member`, which this user is apart of, we stop expanding that group because it matches one of the provided `user_filter` inputs, namely the `group#member` filter. So in this case for this user we only return the computed relation that stems from the `document:example#editor` resolution path.
+
+2. Notice the usage of the relation 'viewer' in the request. This relation is the greatest common denominator of the relations included in the list of permissions in the UI dialog next to each user/subject's name. Here's how to interpret the response and how the Google Docs app would use the response to build the UI:
+
+    - Since `user:maria` is the owner we find resolution paths stemming from `document:example#editor` and `document:example#viewer` thus leading to `document:example#viewer`. The Google Doc Share Dialog chooses to prioritize the highest permission in the UI dropdown boxes, so in this case the UI shows the `owner` relation next to `user:maria`
+
+     - Likewise, since `user:will` is assigned `document:example#editor` which has a computed relationship to `document:example#viewer`. So in this case the UI shows the `editor` relation next to `user:will`
+
+    - `user:andres` can view the parent folder `folder:x`, and thus we find a resolution path to `document:example#viewer` through `folder:x#viewer`. The Google Doc Share Dialog chooses to render the inherited permission `document#viewer` based on this resolution path.
+
+    - The set of users/subjects included in `group:engineering` have editor acces, and thus we find a resolution path stemming from `document:example#viewer` that leads to `document:example#viewer`. The Google Doc Share Dialog chooses to render group memberships such as this as the Google Group and the highest permission that set of subjects has, which is `viewer` in this case.
+
+3. The `user` type wildcard is returned in the `wildcard_types` portion of the response. This special treatment lends nicely to the Google Doc Share example, which has a visually unique treatment.
+
+This response indicates to the Google Docs apps that viewer permissions to `document:example` have been explicitly granted to any object of type `user`, and then the app can use this to implement a link-based sharing mechanism.
+
+### Multiple Target Relations
+
+The ability to specify multiple target relations was considered. This would have enabled developers to display users who had a variety of relations to a specific object. Supporting this would have introduced performance concerns and all the existing endpoints operate on a single-relation basis, which would have made this behavior inconsistent. 
```

Comment:
This would have enabled developers to list users of more than one type that have a specific relation with an object.